src/Common/src/System/LocalAppContext.cs

```diff
@@ -8,6 +8,15 @@ namespace System
 {
     internal partial class LocalAppContext
     {
+        static LocalAppContext()
+        {
+            bool isEnabled;
+            if (AppContext.TryGetSwitch(@"TestSwitch.LocalAppContext.DisableCaching", out isEnabled))
+            {
+                DisableCaching = isEnabled;
+            }
+        }
```

Comment:
Thanks @stephentoub we're actually already getting rid of the static constructor on this pull request #14778. If you could take a look would be great :)